enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -266,35 +259,137 @@ The initial enhancement provides a subset of the options that are available with
-2. Create a Subscription to install OSSM.
-3. Create a ServiceMeshControlPlane to configure OSSM (what if it already exists?; see [Open Questions](#open-questions)).
-4. Create a LoadBalancer-type service (or OSSM may create it; see [Open Questions](#open-questions)).
-5. Create a DNSRecord for the service (see the following subsection).
+This enhancement adds a new controller to create the gatewayclasses CRD.  For
+dev preview, this controller watches for the "GatewayAPI" feature to be enabled
+in the cluster featuregate.  When the feature is enabled, this controller
+installs the Gateway API CRDs, including the gatewayclasses CRD, which enables
+the cluster admin to create a new GatewayClass.
 
 The Ingress Operator effectively owns the Gateway API CRDs.  It installs
 versions of the CRDs that are compatible with the OSSM version that it installs,
 and if the CRDs are already installed, the operator may overwrite them.
 
-When OSSM is installed, it deploys the Istio control-plane and Envoy proxy.  The
-Ingress Operator configures the ServiceMeshControlPlane to enable Gateway API
-support, which means that Istio watches Gateway and HTTPRoute objects and
-configures Envoy accordingly.  Configuring a load balancer and DNS enables
+#### New Controller to Install and Configure OpenShift Service Mesh
+
+The enhancement adds another new controller to reconcile GatewayClasses by
+installing OpenShift Service Mesh (OSSM) when appropriate.  When the cluster
+admin creates a GatewayClass that specifies the
+"openshift.io/gateway-controller" controller name, this controller then does the
+following:
+
+1. Create a Subscription to install OSSM.
+2. Create a ServiceMeshControlPlane to configure OSSM.
+
+When OSSM is installed, it deploys the Istio control-plane.  The Ingress
+Operator configures the ServiceMeshControlPlane to enable Gateway API support,
+which means that Istio watches Gateway and HTTPRoute objects and configures
+Envoy accordingly.  The Ingress Operator also configures the
+ServiceMeshControlPlane to enable Istio's "[Automated Deployment](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#automated-deployment)" feature,
+which means that Istio automatically creates an Envoy proxy deployment and a
+LoadBalancer-type service for each Gateway.  Suppose the cluster admin creates a
+Gateway that specifies a GatewayClass with OpenShift's controller name:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: openshift-ingress
+spec:
+  gatewayClassName: openshift-default
+  # ...
+```
+
+Then OSSM configures Istio and Envoy and creates a LoadBalancer-type service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    gateway.istio.io/managed: openshift.io-gateway-controller
+  name: gateway
+  namespace: openshift-ingress
+  ownerReferences:
+  - apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Even though this is as its done now, this is a bug, it should be v1beta1. Or maybe
```suggestion
  - apiVersion: gateway.networking.k8s.io/<currentversion>
```

Reply:
I made an effort to have this document accurately reflect what you would see on an actual cluster.  How about if I add a footnote that this may change, with a reference to https://issues.redhat.com/browse/OCPBUGS-8681?  Updated in https://github.com/openshift/enhancements/compare/213a342986d10d3ebaaf83ee784ca2867d8b0eb7..dffce6339d368666c5acbb4272e29b8cee3166b2.